### PR TITLE
doing homework10

### PR DESCRIPTION
**Pull request by Hromy-Ann**

### CODE REVIEW

File: src/main/java/org/example/homework10/task2/Employee.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+package org.example.homework10.task2;
+
+public record Employee(String firstName,
+                       String lastName,
+                       String hireDate) {
+    public String firstName() {
```
Comment (alexshamrai) [inline src/main/java/org/example/homework10/task2/Employee.java]: you don't need this getter in the Record

File: src/main/java/org/example/homework10/task2/Employee.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        return firstName;
+    }
+
+    public String lastName() {
```
Comment (alexshamrai) [inline src/main/java/org/example/homework10/task2/Employee.java]: you don't need this getter in the Record

File: src/main/java/org/example/homework10/task2/Employee.java [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+        return lastName;
+    }
+
+    public String hireDate() {
```
Comment (alexshamrai) [inline src/main/java/org/example/homework10/task2/Employee.java]: you don't need this getter in the Record

File: src/main/java/org/example/homework10/task2/Task2.java [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                        '}';
+            }
+        }
+
+        List<CustomEmployee> customEmployees = new ArrayList<>(list.size());
+        for (Employee employee : list) {
+            customEmployees.add(new CustomEmployee(employee));
+        }
+
+        Comparator<CustomEmployee> customEmployeeComparator;
```
Comment (alexshamrai) [inline src/main/java/org/example/homework10/task2/Task2.java]: all this stuff could be done a bit easier
```suggestion
                var comparator = new Comparator<CustomEmployee>() {

            @Override
            public int compare(CustomEmployee o1, CustomEmployee o2) {

                if (sortField.equals("name")) {
                    return o1.fullName.compareTo(o2.fullName);
                }
                return o1.yearsWorked - o2.yearsWorked;
            }
        };
```

and after it you can just use it in `list.sort(comparator);`

**Review by alexshamrai — CHANGES_REQUESTED** [review]

File: src/main/java/org/example/homework10/task2/Employee.java (hunk 1 above)
Comment (Hromy-Ann) [reply src/main/java/org/example/homework10/task2/Employee.java]: removed

File: src/main/java/org/example/homework10/task2/Employee.java (hunk 3 above)
Comment (Hromy-Ann) [reply src/main/java/org/example/homework10/task2/Employee.java]: removed

File: src/main/java/org/example/homework10/task2/Employee.java (hunk 2 above)
Comment (Hromy-Ann) [reply src/main/java/org/example/homework10/task2/Employee.java]: removed

File: src/main/java/org/example/homework10/task2/Task2.java (hunk 4 above)
Comment (Hromy-Ann) [reply src/main/java/org/example/homework10/task2/Task2.java]: I changed 2 comparators in switch to a single one with if branches.

**Review by alexshamrai — APPROVED** [review]

**Pull request merged**